core.go

```diff
@@ -407,6 +417,33 @@ func (c *Core) RemoveLookup(lookup, lookupID, key string) (err error) {
 	return
 }
 
+// Backup will create a full backup of the db into a given directory path
+func (c *Core) Backup(ctx context.Context, dir string, example Value) (err error) {
+	var backup *Core
+	if backup, err = New(c.name, dir, example, c.getRelationships()...); err != nil {
+		return
+	}
+	defer backup.Close()
+
+	dbCopy := make(map[string]Value)
+	if err = c.ForEach(func(key string, val Value) (err error) {
+		dbCopy[key] = val
+		return
+	}); err != nil {
+		return
+	}
+
+	err = backup.Transaction(ctx, func(txn *Transaction) (err error) {
```

Comment:
Is there a more efficient way?

Reply:
Can you clarify?  I just noticed it's probably not best to keep the original value in memory and then write in single transaction (we might be keep 10+gb in memory at a time).  We should write an entry at a time as we're iterating through.

We need to initialize the backup db though because of the exclusive locks.

Reply:
@classic-massok 

Maybe we can batch transactions in queued groups of 1000 or something empirically tuned to balance memory and performance?

Reply:
There is a better way
https://github.com/gdbu/snapshotter/blob/master/frontends/bolt.go#L24